anvil/src/eth/backend/mem/mod.rs

```diff
@@ -751,16 +759,12 @@ impl Backend {
         }).await?
     }
 
-    pub fn call_with_state<D>(
+    fn build_call_env(
         &self,
-        state: D,
         request: EthTransactionRequest,
         fee_details: FeeDetails,
         block_env: BlockEnv,
-    ) -> Result<(Return, TransactOut, u64, State), BlockchainError>
-    where
-        D: DatabaseRef,
-    {
+    ) -> Result<Env, BlockchainError> {
```

Comment:
can this even return an error?
I couldn't find any `?`, so perhaps we just return the `Env`?

Reply:
Definitely! I thought this was the kind of stuff `clippy` would catch...
It's done :)